src/Plugins/SimplnxCore/src/SimplnxCore/Filters/ComputeFeatureBoundsFilter.cpp

```diff
@@ -0,0 +1,209 @@
+  params.linkParameters(k_OutputType_Key, k_MinArrayName_Key, static_cast<ChoicesParameter::ValueType>(to_underlying(ComputeFeatureBounds::OutputDataType::Split)));
+  params.linkParameters(k_OutputType_Key, k_MaxArrayName_Key, static_cast<ChoicesParameter::ValueType>(to_underlying(ComputeFeatureBounds::OutputDataType::Split)));
+  params.linkParameters(k_OutputType_Key, k_UnifiedArrayName_Key, static_cast<ChoicesParameter::ValueType>(to_underlying(ComputeFeatureBounds::OutputDataType::Unified)));
+
+  return params;
+}
+
+//------------------------------------------------------------------------------
+IFilter::VersionType ComputeFeatureBoundsFilter::parametersVersion() const
+{
+  return 1;
+}
+
+//------------------------------------------------------------------------------
+IFilter::UniquePointer ComputeFeatureBoundsFilter::clone() const
+{
+  return std::make_unique<ComputeFeatureBoundsFilter>();
+}
+
+//------------------------------------------------------------------------------
+IFilter::PreflightResult ComputeFeatureBoundsFilter::preflightImpl(const DataStructure& dataStructure, const Arguments& filterArgs, const MessageHandler& messageHandler,
+                                                                   const std::atomic_bool& shouldCancel, const ExecutionContext& executionContext) const
+{
+  auto pOutputTypeValue = filterArgs.value<ChoicesParameter::ValueType>(k_OutputType_Key);
+  auto pFeatureAMPathValue = filterArgs.value<AttributeMatrixSelectionParameter::ValueType>(k_FeatureAMPath_Key);
+  auto pFeatureIdsArrayPathValue = filterArgs.value<ArraySelectionParameter::ValueType>(k_FeatureIdsArrayPath_Key);
+  auto pSelectedGeomPathValue = filterArgs.value<GeometrySelectionParameter::ValueType>(k_SelectedGeometryPath_Key);
+
+  const auto& geom = dataStructure.getDataRefAs<IGeometry>(pSelectedGeomPathValue);
+
+  usize expectedFeatureSize = 0;
+  std::string targetStr = "";
+  switch(geom.getGeomType())
+  {
+  case IGeometry::Type::Image: {
+    expectedFeatureSize = dynamic_cast<const ImageGeom&>(geom).getNumberOfCells();
+    targetStr = "cells";
+    break;
+  }
+  case IGeometry::Type::Triangle: {
+    expectedFeatureSize = dynamic_cast<const TriangleGeom&>(geom).getNumberOfFaces();
+    targetStr = "triangles";
+    break;
+  }
+  case IGeometry::Type::Vertex: {
+    expectedFeatureSize = dynamic_cast<const VertexGeom&>(geom).getNumberOfVertices();
+    targetStr = "vertices";
+    break;
+  }
+  case IGeometry::Type::Edge: {
+    expectedFeatureSize = dynamic_cast<const EdgeGeom&>(geom).getNumberOfEdges();
+    targetStr = "edges";
+    break;
+  }
+  case IGeometry::Type::Quad: {
+    expectedFeatureSize = dynamic_cast<const QuadGeom&>(geom).getNumberOfFaces();
+    targetStr = "faces";
+    break;
+  }
+  default: {
+    return MakePreflightErrorResult(-89473, fmt::format("Unexpected input geometry type. Geometry name {}", geom.getName()));
+  }
+  }
```

Comment:
```suggestion
  usize expectedFeatureSize = geom.getNumberOfCells();
```
`getNumberOfCells` is virtual and calls the correct `getNumberOf` function.